F1WM/Controllers/NewsController.cs

```diff
@@ -49,7 +49,100 @@ public async Task<IActionResult> GetSingle(int id)
 			}
 		}
 
-		[HttpGet("important")]
+        [HttpGet("tag/{id}")]
+        [Produces("application/json", Type = typeof(NewsSummary))]
+        public async Task<IActionResult> GetByTag(int id)
+        {
+            try
+            {
+                var news = await service.GetNewsByTag(id);
+                return (news.Any() ? (IActionResult)Ok(news) : (IActionResult)NotFound());
+            }
+            catch (Exception ex)
+            {
+                logger.LogError(ex);
+                throw ex;
+            }
+        }
+
+        [HttpGet("type/{id}")]
+        [Produces("application/json", Type = typeof(NewsSummary))]
+        public async Task<IActionResult> GetByType(int id)
+        {
+            try
+            {
+                var news = await service.GetNewsByType(id);
+                return (news.Any() ? (IActionResult)Ok(news) : (IActionResult)NotFound());
+            }
+            catch (Exception ex)
+            {
+                logger.LogError(ex);
+                throw ex;
+            }
+        }
+
+        [HttpGet("types")]
+        [Produces("application/json", Type = typeof(NewsType))]
+        public async Task<IEnumerable<NewsType>> GetTypes()
+        {
+            try
+            {
+                return await service.GetNewsTypes();             
+            }
+            catch (Exception ex)
+            {
+                logger.LogError(ex);
+                throw ex;
+            }
+        }
+
+        [HttpGet("tags")]
+        [Produces("application/json", Type = typeof(NewsTag))]
+        public async Task<IEnumerable<NewsTag>> GetTags()
+        {
+            try
+            {
+                return await service.GetNewsTags();
+            }
+            catch (Exception ex)
+            {
+                logger.LogError(ex);
+                throw ex;
+            }
+        }
+
+        [HttpGet("tags/category/{id}")]
+        [Produces("application/json", Type = typeof(NewsTag))]
+        public async Task<IActionResult> GetTagsbyCategory(int id)
+        {
+            try
+            {
+                var tags = await service.GetNewsTagsByCategory(id);
+                return (tags.Any() ? (IActionResult)Ok(tags) : (IActionResult)NotFound());
+            }
+            catch (Exception ex)
+            {
+                logger.LogError(ex);
+                throw ex;
+            }
+        }
+
+        [HttpGet("categories")]
+        [Produces("application/json", Type = typeof(NewsCategory))]
```

Comment:
You don't need a ProducesAttribute here (which specifies wrong type by the way), it can be deduced from the method return type. See GetMany method for example.